tests/test_core.py

```diff
@@ -1,6 +1,7 @@
 import functools
 import math
 import os
+import random
```

Comment:
Instead of importing `random`, why not just do `np.random.rand()`? That will give you a single random number and without needing to import `random` when `np` is available?

Reply:
Right! Noticed it now after carefully reading the doc! Going to fix it now!